scripts/lintcommit.lua

```diff
@@ -0,0 +1,35 @@
+-- Captures shell output and returns it.
+function os.capture(cmd, raw)
+  local f = assert(io.popen(cmd, 'r'))
+  local s = assert(f:read('*a'))
+  f:close()
+  if raw then return s end
+  s = string.gsub(s, '^%s+', '')
+  s = string.gsub(s, '%s+$', '')
+  s = string.gsub(s, '[\n\r]+', ' ')
+  return s
+end
+
+current_branch = os.capture("git branch --show-current")
```

Comment:
`vim.system(['git', 'branch', '--show-current'])`

same for this others. Don't use shell.

Reply:
Are the `[]` meant to be `{}`?  Getting a `E5112: Error while creating lua chunk: scripts/lintcommit.lua:13: unexpected symbol near '['` otherwise.

Reply:
I got it to work with help from @gpanders. I need to know why this change is preferred so I can internalize it and avoid it next time. Is it because of better performance?

Reply:
> I need to know why this change is preferred so I can internalize it and avoid it next time. Is it because of better performance?


it avoids use of the shell, which avoids quoting issues and other shell quirks. see `:help system()`